Marlin/src/lcd/menu/menu.cpp

```diff
@@ -429,10 +429,18 @@ void scroll_screen(const uint8_t limit, const bool is_menu) {
     if (ui.should_draw()) {
       #if ENABLED(BABYSTEP_HOTEND_Z_OFFSET)
         if (!do_probe)
-          MenuEditItemBase::draw_edit_screen(GET_TEXT(MSG_HOTEND_OFFSET_Z), ftostr43sign(hotend_offset[active_extruder].z));
+          #if (Z_PROBE_OFFSET_RANGE_MIN < -9 || Z_PROBE_OFFSET_RANGE_MAX > 9)
+            MenuEditItemBase::draw_edit_screen(GET_TEXT(MSG_HOTEND_OFFSET_Z), ftostr52sign(hotend_offset[active_extruder].z));
+          #else
+            MenuEditItemBase::draw_edit_screen(GET_TEXT(MSG_HOTEND_OFFSET_Z), ftostr54sign(hotend_offset[active_extruder].z));
```

Comment:
I don't quite understand ftostr54sign. How can that handle 9 digits but the others only handle 7?

Is ftostr54sign actually getting cropped to ftostr43sign elsewhere?

Reply:
<= -10.0 no ? 9.025 may be wanted :p

Reply:
Preprocessor doesn't handle float math well so I kept it to closest int since im tossing it together at my desk between other things and not testing atm :)

Reply:
right ok.. anyway, i dont really think the problem is there for the ABL issue..

Reply:
@tpruvot there may still be other leveling problems. This is addressing an issue where you enter an edit screen which cannot handle the present value of the number, and the only way to exit the screen is to save the value shown on the screen, which is incorrect.